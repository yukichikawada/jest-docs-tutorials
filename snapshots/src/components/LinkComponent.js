import React from 'react';

const STATUS = {
  HOVERED: 'hovered',
  NORMAL:  'normal',
};

export default class LinkComponent extends React.Component {
  constructor() {
    super();

    this._onMouseEnter = this._onMouseEnter.bind(this);
    this._onMouseLeave = this._onMouseLeave.bind(this);

    this.state = {
      status: STATUS.NORMAL,
    };
  }

  _onMouseEnter() {
    this.setState({status: STATUS.HOVERED});
  }

  _onMouseLeave() {
    this.setState({status: STATUS.NORMAL});
  }

  render() {
    return (
      <a className={this.state.status}
        id="LinkComponent"
        href={this.props.page || '#'}
        onMouseEnter={this._onMouseEnter}
        onMouseLeave={this._onMouseLeave}
      >
        {this.props.children}
      </a>
    );
  }
}