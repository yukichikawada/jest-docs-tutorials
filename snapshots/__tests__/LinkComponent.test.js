
import React from 'react';
import LinkComponent from '../src/components/LinkComponent';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<LinkComponent page="http://www.instagram.com">Instagram</LinkComponent>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('changes the class on hover', () => {
  const component = renderer.create(
    <LinkComponent page="http://www.facebook.com">Facebook</LinkComponent>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the cb
  tree.props.onMouseEnter();
  // re-render
  tree = component.toJSON();
  // why does the reassignment of tree result in a component
  // with the className hovered?
  // why isn't the order of line 21 & 23 reversed?
  expect(tree).toMatchSnapshot();
  // manually trigger cb
  tree.props.onMouseLeave();
  // re-render
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});