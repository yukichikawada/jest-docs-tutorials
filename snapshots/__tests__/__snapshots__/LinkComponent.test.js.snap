// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the class on hover 1`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  id="LinkComponent"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`changes the class on hover 2`] = `
<a
  className="hovered"
  href="http://www.facebook.com"
  id="LinkComponent"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`changes the class on hover 3`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  id="LinkComponent"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`renders correctly 1`] = `
<a
  className="normal"
  href="http://www.instagram.com"
  id="LinkComponent"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Instagram
</a>
`;
